pkg/vips/image.go

```diff
@@ -24,33 +24,38 @@ type ImageRef struct {
 	buf []byte
 }
 
+// LoadOptions contains optional configuration for the image loading.
+type LoadOptions struct {
+	Access Access
+}
+
 // LoadImage loads an ImageRef from the given reader
-func LoadImage(r io.Reader) (*ImageRef, error) {
+func LoadImage(r io.Reader, opts *LoadOptions) (*ImageRef, error) {
```

Comment:
Hm, I think i'd prefer the functor pattern, then there won't be any upstream changes needed at callsites. Thoughts?

```suggestion
func LoadImage(r io.Reader, opts ...LoadOption) (*ImageRef, error) {
```

Then it'd look like:
```golang
LoadImage(r, WithAccessMode(SequentialAccess))
```

Or something like that.

Reply:
Yes, you're right.
I will change it to functors.